File: src/main/java/kr/co/vacgom/api/babymanager/BabyManager.java
package kr.co.vacgom.api.babymanager;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import kr.co.vacgom.api.baby.Baby;
import kr.co.vacgom.api.global.entity.BaseEntity;
import kr.co.vacgom.api.user.User;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Comment;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Table(name = "TB_BABY_MANAGER")
public class BabyManager extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(
            name = "baby_manager_id",
            nullable = false,
            updatable = false
    )
    @Comment("[Not Null] 아기 매니저 id")
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "manager_id", nullable = false)
    @Comment("[Not Null] 아기 매니저 id")
    private User manager;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "baby_id", nullable = false)
    @Comment("[Not Null] 아기 id")
    private Baby baby;

    @Column(nullable = false)
    @Comment("[Not Null] 대표 돌보미 여부")
    private Boolean isAdmin;

    public BabyManager(User manager, Baby baby, Boolean isAdmin) {
        this.manager = manager;
        this.baby = baby;
        this.isAdmin = isAdmin;
    }
}

File: src/main/kotlin/kr/co/vacgom/api/babymanager/domain/BabyManager.kt
package kr.co.vacgom.api.babymanager.domain

import jakarta.persistence.*
import kr.co.vacgom.api.baby.domain.Baby
import kr.co.vacgom.api.global.common.domain.BaseTimeEntity
import kr.co.vacgom.api.global.util.UuidCreator
import kr.co.vacgom.api.user.domain.User
import org.hibernate.annotations.Comment
import java.util.*

@Entity
@Table(name = "TB_BABY_MANAGER")
class BabyManager(
    id: UUID = UuidCreator.create(),
    user: User,
    baby: Baby,
    isAdmin: Boolean = false,
) : BaseTimeEntity() {
    @Id
    @Column(name = "baby_manager_id", nullable = false, updatable = false)
    @Comment("[Not Null] 아기 매니저 id")
    var id: UUID = id
        protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "manager_id", nullable = false)
    @Comment("[Not Null] 아기 매니저 id")
    var user: User = user
        protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "baby_id", nullable = false)
    @Comment("[Not Null] 아기 id")
    var baby: Baby = baby
        protected set

    @Column(nullable = false)
    @Comment("[Not Null] 대표 돌보미 여부")
    var isAdmin: Boolean = isAdmin
        protected set
}
